src/Symfony/Component/FeatureFlag/DataCollector/FeatureCheckerDataCollector.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag\DataCollector;
+
+use Symfony\Component\FeatureFlag\FeatureRegistry;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+use Symfony\Component\HttpKernel\DataCollector\LateDataCollectorInterface;
+
+final class FeatureCheckerDataCollector extends DataCollector implements LateDataCollectorInterface
+{
+    private array $isEnabledLogs = [];
+    private array $valueLogs = [];
+
+    public function __construct(
+        private readonly FeatureRegistry $featureRegistry,
+    )
+    {
+        $this->data = [
+            'features' => [],
+        ];
+    }
+
+    public function collect(Request $request, Response $response, \Throwable $exception = null): void
+    {
+    }
+
+    public function collectIsEnabled(string $featureName, bool $result): void
+    {
+        $this->isEnabledLogs[$featureName] = $result;
+    }
+
+    public function collectValue(string $featureName, mixed $value): void
+    {
+        $this->valueLogs[$featureName] = $value;
+    }
+
+    public function getName(): string
+    {
+        return 'feature_flag';
+    }
+
+    public function reset(): void
```

Comment:
Is is needed as the parent method would reset `$this->data`, and the `data['features']` is initialized in lateCollect ?

Reply:
Good catch. I updated the DataCollector to call the parent's method and reset its own properties.